### ISSUE TITLE
Protocol matching doesn't enforce `ClassVar`

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#4156]
The following code should produce a type violation but doesn't.

```python
from typing import ClassVar, Protocol

class A(Protocol):
    x: ClassVar[int]

class B:
    x: int

b = B()

a: A = b
```

**Comment by erictraut** [comment]
This is included in pyright 1.1.279, which I just published. It will also be included in a future version of pylance.
